From da3e63d47415b129d68ba7aebc2b27cb3e86e618 Mon Sep 17 00:00:00 2001
From: Khalvat-M <mmba313@gmail.com>
Date: Tue, 25 Oct 2022 06:41:39 +0330
Subject: [PATCH 2/2] Revert "config: Remove TARGET_DISABLE_POSTRENDER_CLEANUP"

This reverts commit fa5bc45a95fae4cbd68b3d19265f5f1e799792e2.

(cherry picked from commit 513c7d32258c9eaf78cb43fe42e22d6ffae6bd82)
---
 build/soong/Android.bp     | 17 +++++++++++++++++
 config/BoardConfigSoong.mk |  2 ++
 2 files changed, 19 insertions(+)

diff --git a/build/soong/Android.bp b/build/soong/Android.bp
index e4bd63ec..ef1dfec3 100644
--- a/build/soong/Android.bp
+++ b/build/soong/Android.bp
@@ -61,6 +61,23 @@ cc_library_headers {
 }
 
 // Target platform agnostic config modules
+soong_config_module_type {
+    name: "disable_postrender_cleanup",
+    module_type: "cc_defaults",
+    config_namespace: "lineageGlobalVars",
+    bool_variables: ["disable_postrender_cleanup"],
+    properties: ["cppflags"],
+}
+
+disable_postrender_cleanup {
+    name: "disable_postrender_cleanup_defaults",
+    soong_config_variables: {
+        disable_postrender_cleanup: {
+            cppflags: ["-DDISABLE_POSTRENDER_CLEANUP"],
+        },
+    },
+}
+
 soong_config_module_type {
     name: "aapt_version_code",
     module_type: "java_defaults",
diff --git a/config/BoardConfigSoong.mk b/config/BoardConfigSoong.mk
index 776e1da0..1ffeb646 100644
--- a/config/BoardConfigSoong.mk
+++ b/config/BoardConfigSoong.mk
@@ -33,6 +33,7 @@ SOONG_CONFIG_lineageGlobalVars += \
     bootloader_message_offset \
     camera_override_format_from_reserved \
     gralloc_handle_has_custom_content_md_reserved_size \
+    disable_postrender_cleanup \
     gralloc_handle_has_reserved_size \
     gralloc_handle_has_ubwcp_format \
     target_health_charging_control_charging_path \
@@ -76,6 +77,7 @@ endif
 # Soong bool variables
 SOONG_CONFIG_lineageGlobalVars_camera_override_format_from_reserved := $(TARGET_CAMERA_OVERRIDE_FORMAT_FROM_RESERVED)
 SOONG_CONFIG_lineageGlobalVars_gralloc_handle_has_custom_content_md_reserved_size := $(TARGET_GRALLOC_HANDLE_HAS_CUSTOM_CONTENT_MD_RESERVED_SIZE)
+SOONG_CONFIG_lineageGlobalVars_disable_postrender_cleanup := $(TARGET_DISABLE_POSTRENDER_CLEANUP)
 SOONG_CONFIG_lineageGlobalVars_gralloc_handle_has_reserved_size := $(TARGET_GRALLOC_HANDLE_HAS_RESERVED_SIZE)
 SOONG_CONFIG_lineageGlobalVars_gralloc_handle_has_ubwcp_format := $(TARGET_GRALLOC_HANDLE_HAS_UBWCP_FORMAT)
 SOONG_CONFIG_lineageGlobalVars_uses_egl_display_array := $(TARGET_USES_EGL_DISPLAY_ARRAY)
-- 
2.48.1

